src/crustls.h

```diff
@@ -223,6 +249,45 @@ struct rustls_client_config_builder *rustls_client_config_builder_new(void);
  */
 const struct rustls_client_config *rustls_client_config_builder_build(struct rustls_client_config_builder *builder);
 
+/**
+ * Set a custom server certificate verifier.
+ *
+ * The userdata pointer must stay valid until (a) all sessions created with this
+ * config have been freed, and (b) the config itself has been freed.
+ * The callback must not capture any of the pointers in its
+ * rustls_verify_server_cert_params.
+ *
+ * The callback must be safe to call on any thread at any time, including
+ * multiple concurrent calls. So, for instance, if the callback mutates
+ * userdata (or other shared state), it must use synchronization primitives
+ * to make such mutation safe.
+ *
+ * The callback receives certificate chain information as raw bytes.
+ * Currently this library offers no functions for C code to parse the
+ * certificates, so it's only possible to implement verifiers that either
+ * (a) always succeed (or fail), or (b) compare the certificates against
+ * static bytes. We plan to export parsing code in the future to make it
+ * possible to implement other strategies.
```

Comment:
IIUC, the "raw bytes" will point to a `[rustls::Certificate]`. I think that rather than commit to extending `crustls`'s API surface with a rich representation of an X.509 certificate, you should provide this callback the DER encoded certificate chain, because then the client can bring whatever library they want (such as OpenSSL, or maybe a project like `curl` already has a bunch of code for working with certs and won't want to adopt a new interface), parse the certificates and do whatever they want without `crustls` needing to specifically provide an API affordance to enable their use case. Note that I'm not arguing for committing to that API contract now, but rather removing the promise to eventually provide a certificate API in `crustls`.